NET-specifics/data-binding.md

```diff
@@ -1,2 +1,213 @@
+}
+```
+
+## Screenbinder
+
+The next step after creating this is to define a screenbinder that would take care of the binding for us. It inherits from the class OSPSuite.DataBinding.ScreenBinder<TObjectType> defined in [OSPSuite.DataBinding](https://github.com/Open-Systems-Pharmacology/OSPSuite.DataBinding/blob/master/src/OSPSuite.DataBinding/ScreenBinder.cs). 
+
+So moving back to the view, we define a screenbinder for this newly created DTO:
+
+```
+private readonly ScreenBinder<MyValueDTO> _screenBinder;
+```
+
+Since our view is extending `BaseView` (or even as part of the `IView` interface), we have a base virtual function `void InitializeBinding()`. This function should take care of the initializing of the binding, so in our case:
+
+```
+public override void InitializeBinding()
+{
+      base.InitializeBinding();
+      _screenBinder.Bind(x => x.MyValue).To(myValueTextEdit); 
+}
+```
+
+In order to bind to the actual DTO object, we need a `void BindTo(MyValueDTO myValueDTO)` function as part of the interface of the view. This function can then be called from the presenter to provide the DTO object for the actual binding.
+
+So in the view we will have the implementation of this function:
+
+```
+public void BindTo(MyValueDTO myValueDTO)
+{
+      _screenBinder.BindToSource(myValueDTO);
+}
+```
+
+Like this, the binding is complete. Similarly we could expand the DTO with more fields and use it to bind to various UI elements. In the case we had f.e. a form that would consist of a drop-down, a color selector, and two TextEdits for value input, we would usually create a single DTO that would encapsulate all those values and bind to the separate visual elements. An example of such a more complex binding can be found in [ParameterIdentificationConfigurationView.cs](https://github.com/Open-Systems-Pharmacology/OSPSuite.Core/blob/develop/src/OSPSuite.UI/Views/ParameterIdentifications/ParameterIdentificationConfigurationView.cs), both in the initialization and the binding segments.
+
+## Validation
+
+Going back to the simple example that we are presenting here. If we want to add validation to the values that come as input by the user for the TextEdit, we need to implement in the DTO the `IValidatable` interface. Then we have to define the rule that we need validated. Let's say that we need the float value that we are reading to be greater than 2.
+
+
+```
+public class MyValueDTO : IValidatable
+{
+public float MyValue { get; set; }
+
+public IBusinessRuleSet Rules => AllRules.Default;
+
+private static class AllRules
+{
+      private static IBusinessRule myValueGreaterThanTwo
+      {
+      get
+      {
+            return CreateRule.For<MyValueDTO>()
+            .Property(item => item.MyValue)
+            .WithRule((x, value) => value > 2.0)
+            .WithError("Value must be greater than 2");
+      }
+      }
+
+      public static IBusinessRuleSet Default => new BusinessRuleSet(myValueGreaterThanTwo);
+}
+}
+```
```

Comment:
Too bad the indentation can't be done automatically. But it should be fixed up before merging

Reply:
done